pettingzoo/mpe/scenarios/group_spread.py

```diff
@@ -48,44 +47,37 @@ def make_world(self, groups, num_colors=None, reward_per_group=True, randomise_a
+
+        # assign one of the available colors to a group
+        color_idxs = np.random.choice(self.num_colors, len(self.groups), replace=False)
+
+        # assign group / color properties to agents
         for i, agent in zip(self.group_indices, world.agents):
             agent.color = self.colors[color_idxs[i]]
             agent.color_idx = color_idxs[i]
             agent.group = i
 
-        # random properties for landmarks
-        if self.shuffle_obs_per_agent:
-            for i, landmark in zip(self.group_indices, world.landmarks):
-                landmark.color = self.colors[color_idxs[i]]
-                landmark.color_idx = color_idxs[i]
-                landmark.group = i
-            # assign each agent a landmark and agent ordering that wil determine order in the agent obs
-            # this allows entities to be shuffled per agent while keeping consistent for the episode
-            for agent in world.agents:
-                landmark_obs_order = np.arange(len(world.landmarks))
-                random.shuffle(landmark_obs_order)
-                agent.landmark_obs_order = landmark_obs_order
-                agent_obs_order = np.arange(len(world.agents))
-                random.shuffle(agent_obs_order)
-                agent.agent_obs_order = agent_obs_order
-        else:
-            # assign landmark groups randomly as landmarks won't be shuffled in obs
-            landmarks_ids = np.arange(len(world.landmarks))
-            np_random.shuffle(landmarks_ids)
-            for group_id, land_id in zip(self.group_indices, landmarks_ids):
-                world.landmarks[land_id].color = self.colors[color_idxs[group_id]]
-                world.landmarks[land_id].color_idx = color_idxs[group_id]
-                world.landmarks[land_id].group = group_id
-
-        # set random initial states
+        # assign properties to landmarks
+        landmarks_ids = np.random.choice(self.num_landmarks, self.num_landmarks, replace=False)
+        for group_id, land_id in zip(self.group_indices, landmarks_ids):
+            world.landmarks[land_id].color = self.colors[color_idxs[group_id]]
+            world.landmarks[land_id].color_idx = color_idxs[group_id]
+            world.landmarks[land_id].group = group_id
+
+        # assign each agent a landmark and agent ordering that wil determine order in the agent obs
+        # this allows entities to be shuffled per agent while keeping consistent for the episode
+        for agent in world.agents:
+            if self.shuffle_obs_per_agent:
+                agent.landmark_obs_order = np.random.choice(self.num_landmarks, self.num_landmarks, replace=False)
+                agent.agent_obs_order = np.random.choice(self.num_agents, self.num_agents, replace=False)
+            else:
+                agent.landmark_obs_order = np.arange(self.num_landmarks)
+                agent.agent_obs_order = np.arange(self.num_agents)
```

Comment:
Now always setting this variable, just depending on the shuffling value and setting it to the arange arrays to have the default behaviour if `shuffle_obs_per_agent=False`.

Reply:
(Just for cleaner code in obs construction later)